### PR TITLE
#15 Implement ToolCommandlet for Cobigen

### LINKED ISSUE #15: Implement ToolCommandlet for Cobigen
Create `ToolCommandlet` for Cobigen
Old commandlet:
https://github.com/devonfw/ide/blob/master/scripts/src/main/resources/scripts/command/cobigen

### PR DESCRIPTION
**Pull request by CREITZ25**
closes #15 Initial creation of commandlet Cobigen

### CODE REVIEW

**Review by MattesMrzik — COMMENTED** [review]
Seems good to me. One thing though: in the [bash version of this commandlet](https://github.com/devonfw/ide/blob/bd1435a067ad7db384bc601f9d8e701be419fbf4/scripts/src/main/resources/scripts/command/cobigen#L33) `mvn` is installed as prerequisite and `cobigen` is installed via `doUpgradeMavenArtifact`. This seems to differ to the installation done here.

**Comment by CREITZ25** [comment]
> Seems good to me. One thing though: in the [bash version of this commandlet](https://github.com/devonfw/ide/blob/bd1435a067ad7db384bc601f9d8e701be419fbf4/scripts/src/main/resources/scripts/command/cobigen#L33) `mvn` is installed as prerequisite and `cobigen` is installed via `doUpgradeMavenArtifact`. This seems to differ to the installation done here.

The CobiGen URL Updater has already been implemented. This means that the installation of cobigen can also be implemented using the standard download approach.

File: cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+public class Cobigen extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Cobigen(IdeContext context) {
+
+    super(context, "cobigen", Set.of(TAG_CLOUD, TAG_IAC));
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java:22]: Cobigen is a codegenerator mainly for Java but also for TypeScript or potentially C#/.NET or other languages.
Meanwhile the PR #166 was merged so the Tags are not Strings anymore so this code needs to be updated.

File: cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  @Override
+  public boolean install(boolean silent) {
+
+    getCommandlet(Java.class).install();
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java:28]: @jan-vcapgemini is that sufficient? IMHO cobigen CLI also requires `mvn` so we could replace `Java` with `Mvn` here (as `Mvn` already depends on `Java`).

Comment (jan-vcapgemini) [reply cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java:28]: As @MattesMrzik  already mentioned, I agree that we need to replace Java with Mvn here.

File: cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java (hunk 1 above)
Comment (CREITZ25) [reply cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java]: Changed to Tag.GENERATOR

File: cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java (hunk 2 above)
Comment (CREITZ25) [reply cli/src/main/java/com/devonfw/tools/ide/tool/cobigen/Cobigen.java]: Changed from Java.class to Mvn,class

**Pull request merged**